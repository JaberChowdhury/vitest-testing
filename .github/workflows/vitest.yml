name: Run Vitest Tests

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      # Step 1: Check out the repository
      - name: Checkout code
        uses: actions/checkout@v3

      # Step 2: Set up Node.js with pnpm caching
      - name: Set up Node.js
        uses: actions/setup-node@v3
        with:
          node-version: 18
          cache: "pnpm"

      # Step 3: Install pnpm manually (bypassing Corepack)
      - name: Install pnpm
        run: npm install -g pnpm

      # Step 4: Ensure pnpm is added to PATH
      - name: Add pnpm to PATH
        run: echo "$(npm root -g)/pnpm" >> $GITHUB_PATH

      # Step 5: Verify pnpm installation and check PATH
      - name: Verify pnpm installation
        run: |
          echo "pnpm version:"
          pnpm --version
          echo "pnpm location:"
          which pnpm
          echo "Current PATH:"
          echo $PATH

      # Step 6: Install dependencies
      - name: Install dependencies
        run: pnpm install

      # Step 7: Run Vitest tests
      - name: Run tests
        run: pnpm test
