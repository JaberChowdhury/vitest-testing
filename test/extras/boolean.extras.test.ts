import { describe, expect, test } from "vitest";

describe("Boolean extras", () => {
  //  Truthy and Falsy Values
  test("value is truthy", () => {
    const truthyValue = "hello";
    expect(truthyValue).toBeTruthy();
  });

  test("value is falsy", () => {
    const falsyValue = "";
    expect(falsyValue).toBeFalsy();
  });
});
describe("Boolean extras", () => {
  // Negating Boolean Checks
  test("value is not true", () => {
    const isFalse = false;
    expect(isFalse).not.toBe(true);
  });

  test("value is not false", () => {
    const isTrue = true;
    expect(isTrue).not.toBe(false);
  });
});
describe("Boolean extras", () => {
  // Boolean Expressions
  test("boolean expression is true", () => {
    const a = 5;
    const b = 10;
    expect(a < b).toBe(true);
  });

  test("logical AND operation", () => {
    const isTrue = true;
    const isAlsoTrue = true;
    expect(isTrue && isAlsoTrue).toBe(true);
  });
});
describe("Boolean extras", () => {
  // Checking for Null or Undefined
  test("value is null", () => {
    const nullValue = null;
    expect(nullValue).toBeNull();
  });

  test("value is undefined", () => {
    const undefinedValue = undefined;
    expect(undefinedValue).toBeUndefined();
  });
});
describe("Boolean extras", () => {
  // Custom Boolean Logic
  test("custom boolean logic", () => {
    const isEven = (num) => num % 2 === 0;
    expect(isEven(4)).toBeTruthy();
    expect(isEven(5)).toBeFalsy();
  });
});
describe("Boolean extras", () => {
  // Boolean Return Values from Functions
  function isPositive(num) {
    return num > 0;
  }

  test("function returns true for positive numbers", () => {
    expect(isPositive(10)).toBe(true);
  });

  test("function returns false for non-positive numbers", () => {
    expect(isPositive(-5)).toBe(false);
  });
});
describe("Boolean extras", () => {
  //  Snapshot Testing for Booleans
  test("boolean snapshot", () => {
    const isTrue = true;
    expect(isTrue).toMatchSnapshot();
  });
});
describe("Boolean extras", () => {
  // Testing a Function That Returns a Boolean
  function isPalindrome(str) {
    const cleanedStr = str.replace(/[\W_]/g, "").toLowerCase();
    return cleanedStr === cleanedStr.split("").reverse().join("");
  }

  test("isPalindrome returns true for palindromes", () => {
    expect(isPalindrome("racecar")).toBe(true);
    expect(isPalindrome("A man, a plan, a canal, Panama")).toBe(true);
  });

  test("isPalindrome returns false for non-palindromes", () => {
    expect(isPalindrome("hello")).toBe(false);
    expect(isPalindrome("Vitest")).toBe(false);
  });
});
