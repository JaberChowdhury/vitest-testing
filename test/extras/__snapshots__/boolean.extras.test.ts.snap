// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Boolean extras > boolean snapshot 1`] = `true`;

exports[`Boolean extras > boolean snapshot 2`] = `true`;
